### DISCUSSION TITLE
rate limit - what is the maximum and minimum  time we need to wait before making next call

### DISCUSSION THREAD

**Discussion by desh-per** [community/community#36949] [Apps, API and Webhooks]
what is the maximum and minimum  time we need to wait before making next call ?

**Comment by ossanna16** [comment]
Hi there @desh-per and welcome to our community 😄. Can you please elaborate on your question and provide more detail so we can help?

**Comment by desh-per** [comment]
HI @ossanna16 
As per the doc the rate limit for free account is 5000 request per hour
![image](https://user-images.githubusercontent.com/113818716/198192795-8ee93f93-1183-41b6-bcea-1a8dd0fc3fde.png)
I tried to exceed the rate limit and i observed that i had to wait for 30 minutes to make next API call as per the header "x-ratelimit-reset" 
So i wanted to know if i exceed the rate limit  the idle time to wait is 30 minutes always or can it be less than 30 minutes also

**Reply by ossanna16** [reply]
Thank you for providing more details!

**Reply by desh-per** [reply]
HI @ossanna16  
any update on this ?

**Reply by ossanna16** [reply]
Hey @ghostinhershell and @timrogers! Are you able to help with this question?

**Reply by ghostinhershell** [reply]
Thanks for the ping @ossanna16!

@desh-per, GitHub's API has several layers of limits, therefore different types of API requests to GitHub.com are subject to different rate limits (see https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting). As this is an abuse protection measure, it is based on multiple input variables. Ultimately, the time listed under the x-ratelimit-reset header will let you know when your current rate limit window resets.
